gnocchi/indexer/alembic/versions/18fff4509e3e_create_column_for_truncate_inactive_metrics_process.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2015 OpenStack Foundation
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+"""create metric truncation status column
+
+Revision ID: 18fff4509e3e
+Revises: 04eba72e4f90
+Create Date: 2024-04-24 09:16:00
+
+"""
+import datetime
+
+from alembic import op
+from sqlalchemy.sql import func
+
+import sqlalchemy
+
+# revision identifiers, used by Alembic.
+revision = '18fff4509e3e'
+down_revision = '04eba72e4f90'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column(
+        "metric", sqlalchemy.Column(
+            "needs_raw_data_truncation", sqlalchemy.Boolean,
+            nullable=False, default=True,
+            server_default=sqlalchemy.sql.true()))
```

Comment:
If we set to true, the clean up will not run at least once for all "inactive" metrics.  I mean, metrics that do not receive any datapoints anymore. Therefore, they would not be checked until we update the archive policy backwindow. That is why it is default is true. This will guarantee that we run once for "inactive" metrics as soon as this patch is applied in a production setup.

Yes, the result of this patch is smaller disk usage, depending on the context of the setup. Even though this seems trivial, we, for instance, managed to reduce more than 200GiB of not needed data from a production setup with this patch.

Reply:
sorry, i got confused. i was thinking this was just a flag to enable functionality in general but it's to tell janitor whether it needs to do cleanup.

since we default to true, do we need to include any special steps for upgrade or it should just work (with janitor doing a lot of work initially)?